tests/php/ORM/DataListTest.php

```diff
@@ -2254,172 +2256,150 @@ private function chunkTester(
     /**
      * @dataProvider provideEagerLoadRelations
      */
-    public function testEagerLoadRelations(string $iden, array $eagerLoad, int $expected): void
+    public function testEagerLoadRelations(array $eagerLoad, int $expectedCount): void
     {
         $this->createEagerLoadData();
         $dataList = EagerLoadObject::get()->eagerLoad(...$eagerLoad);
         list($results, $selectCount) = $this->iterateEagerLoadData($dataList);
-        $this->assertSame($this->expectedEagerLoadData(), $results, $iden);
-        $this->assertSame($expected, $selectCount, $iden);
+        $expectedResults=$this->expectedEagerLoadData();
+        $this->assertSame(sort($expectedResults), sort($results));
+        $this->assertSame($expectedCount, $selectCount);
```

Comment:
It seems like this should be all that's needed.... what are all the other changes for? (note that I haven't taken much time to try to understand it - I'm assuming you explaining it will be faster as I've not looked at this test code at all and it seems there's quite a lot going on)

Reply:
I think using the dataprovider array key to identify the test cases is a generally better syntax so I'm updated those.

But I'm not fuss either way. I can revert that back if it's a problem.

Reply:
If you want to keep those changes, please separate them into a _separate commit_ in this PR so I can review them independently of the other changes.

Reply:
Change has been split into two commits.